packages/sync/src/Actions.php

```diff
@@ -348,6 +348,12 @@ public static function do_initial_sync() {
 			return false;
 		}
 
+		// Don't start new sync if a full sync is in process.
+		$full_sync_module = \Jetpack_Sync_Modules::get_module( 'full-sync' );
+		if ( $full_sync_module && ( $full_sync_module->is_started() && ! $full_sync_module->is_finished() ) ) {
+			return false;
```

Comment:
Could we also have a unit test to verify that this functionality works as expected?

Reply:
> It also seems like the existing tests are failing right now.

Fixed with the update to the deprecated class.